flake8_trio/visitors/visitor91x.py

```diff
@@ -394,6 +493,8 @@ def visit_While(self, node: cst.While | cst.For):
         self.loop_state = LoopState()
         self.infinite_loop = self.body_guaranteed_once = False
 
+    visit_For = visit_While
```

Comment:
ooooops, pretty nasty line to have forgotten about. Added a test that would've caught it.